cpp/src/community/k_truss_impl.cuh

```diff
@@ -214,121 +409,501 @@ k_truss(raft::handle_t const& handle,
-    update_edge_src_property(
-      handle, cur_graph_view, out_degrees.begin(), edge_src_out_degrees.mutable_view());
-    update_edge_dst_property(
-      handle, cur_graph_view, out_degrees.begin(), edge_dst_out_degrees.mutable_view());
-
-    rmm::device_uvector<vertex_t> srcs(0, handle.get_stream());
-    rmm::device_uvector<vertex_t> dsts(0, handle.get_stream());
-
-    edge_weight_view =
-      edge_weight ? std::make_optional((*edge_weight).view())
-                  : std::optional<edge_property_view_t<edge_t, weight_t const*>>{std::nullopt};
-    if (edge_weight_view) {
-      std::tie(srcs, dsts, wgts) = extract_transform_e(
-        handle,
-        cur_graph_view,
-        edge_src_out_degrees.view(),
-        edge_dst_out_degrees.view(),
-        *edge_weight_view,
-        extract_low_to_high_degree_weighted_edges_t<vertex_t, weight_t, edge_t>{});
-    } else {
-      std::tie(srcs, dsts) =
-        extract_transform_e(handle,
-                            cur_graph_view,
-                            edge_src_out_degrees.view(),
-                            edge_dst_out_degrees.view(),
-                            edge_dummy_property_t{}.view(),
-                            extract_low_to_high_degree_edges_t<vertex_t, edge_t>{});
-    }
+  // Masking edges not part of the DODG
+  edgelist_dodg.insert(srcs.begin(), srcs.end(), dsts.begin());
 
-    if constexpr (multi_gpu) {
-      std::tie(srcs, dsts, wgts, std::ignore, std::ignore, std::ignore) =
-        detail::shuffle_ext_vertex_pairs_with_values_to_local_gpu_by_edge_partitioning<vertex_t,
-                                                                                       edge_t,
-                                                                                       weight_t,
-                                                                                       int32_t>(
-          handle, std::move(srcs), std::move(dsts), std::move(wgts), std::nullopt, std::nullopt);
-    }
-
-    std::optional<rmm::device_uvector<vertex_t>> tmp_renumber_map{std::nullopt};
-
-    std::tie(*modified_graph, edge_weight, std::ignore, std::ignore, tmp_renumber_map) =
-      create_graph_from_edgelist<vertex_t, edge_t, weight_t, edge_t, int32_t, false, multi_gpu>(
-        handle,
-        std::nullopt,
-        std::move(srcs),
-        std::move(dsts),
-        std::move(wgts),
-        std::nullopt,
-        std::nullopt,
-        cugraph::graph_properties_t{false /* now asymmetric */, cur_graph_view.is_multigraph()},
-        true);
+  cugraph::transform_e(
+    handle,
+    cur_graph_view,
+    edgelist_dodg,
+    cugraph::edge_src_dummy_property_t{}.view(),
+    cugraph::edge_dst_dummy_property_t{}.view(),
+    cugraph::edge_dummy_property_t{}.view(),
+    [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+      return true;
+    },
+    dodg_mask.mutable_view(),
+    false);
```

Comment:
Should we really extract `edgelist_dodg` and call `transform_e`?

Can't we combine the `extract_transform_e` and `transform_e` calls?

Call `transform_e` with `edge_src_out_degrees.view()` & `edge_dst_out_degrees.view()`?